segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java

```diff
@@ -979,8 +962,7 @@ private Void handleException(long requestId, String segment, long offset, String
             log.info(requestId, "Segment '{}' already exists and cannot perform operation '{}'.",
                      segment, operation);
             invokeSafely(connection::send, new SegmentAlreadyExists(requestId, segment, clientReplyStackTrace), failureHandler);
-
-        } else if (u instanceof StreamSegmentNotExistsException) {
```

Comment:
is this change required?

Reply:
`TableStore.getInfo` API can throw 2 types of exceptions:
StreamSegmentNotExistsException
BadSegmentTypeException

So both need to be handled.

Reply:
BadSegmentTypeException would be an unexpected exception for any request by the Controller.

Reply:
An "Exception" is not expected behavior, but if there is any possibility that it could be thrown we need to handle that. What concern do you see with adding that check?

Reply:
`WireCommands.OperationUnsupported` seems more apt for scenarios where Table operations are attempted to be performed over a Stream segment.

Reply:
for e.g: Invoking deleteTableSegment() on a Stream Segment should not return a NoSuchSegment as it would imply the tablesegment is already deleted.

Reply:
`BadSegmentTypeException` will be thrown if you try to get info on a non-table segment (i.e., stream segment) using the `GetTableSegmentInfo` API. If you encounter it, it's likely a code bug and should be handled as "Precondition failed/Illegal Argument" - it shouldn't be retried as it is not an internal server error or transient error.

So I would find it misleading to return NoSuchSegment if you encounter that one. The segment does exist, but it is of the wrong type.

Reply:
For `WireCommand.GetTableSegmentInfo` Controller does not do retries on getting a `NoSuchSegment` response but just fails the request. 
From Controller's pov "StreamSegmentNotExistsException" and "BadSegmentTypeException" both mean a TableSegment by that name does not exist on SS and so returning a "NoSuchSegment" would be the right thing to do.

Reply:
I have removed the check for `BadSegmentTypeException` since it does not need to be added by this PR.

Reply:
A `BadSegmentTypeException` would cause the Segmentstore to close an exception and throw an `IllegalStateException`. Could you please create an issue for this?